t/issue/1508.t

```diff
@@ -0,0 +1,165 @@
+
+=head1 NAME 
+
+issue/1508.t  - Check that the `needs()` function correctly propogates run-time task arguments
+
+=head1 DESCRIPTION
+
+Check that the `needs()` function does indeed correctly propogate 
+run-time task arguments (%params and @args) from the calling task down to the "needed" tasks.
+
+=head1 DETAILS
+
+  * AUTHOR / DATE : [tabulon]@[2021-09-26]
+  * RELATES-TO    : [github issue #1508](https://github.com/RexOps/Rex/issues/1508#issue-1007457392)
+  * INSPIRED from : t/needs.t
+
+=cut
+
+use Test::More;
+use Rex::Commands;
+
+{
+
+  package T; # Helper package (for cutting down boilerplate in tests)
+  use Storable;
+
+  sub track_taskrun {
```

Comment:
If I understand correctly this subroutine saves the arguments passed to it in a file via Storable. If that's true, would it be better to call it something like `save_arguments`?

Reply:
If you wish so...

It also saved the arguments, yes. 
But its principal function is to register the fact that a given task has just been executed (also saving the run-time arguments, in our case).

Reply:
I think by now I better understand the original name. This is a subroutine that tracks task execution (~run).